src/components/App.js

```diff
@@ -1,16 +1,199 @@
-import React from 'react';
+import React from "react";
+import Search from "./Search";
+import Results from "./Results";
+import Pages from "./Pages";
+import Favourites from "./Favourites";
+import Header from "./Header";
 
 class App extends React.Component {
-  constructor(){
+  constructor() {
     super();
+    this.state = {
+      searchTerm: "",
+      movieArray: [],
+      pagesObject: {},
+      currentPage: 1,
+      favouritesArray: [],
+      apiKey: "e2c4cd31"
+    };
+    this.receiveSearchTerm = this.receiveSearchTerm.bind(this);
+    this.fetchMovies = this.fetchMovies.bind(this);
+    this.receiveCurrentPage = this.receiveCurrentPage.bind(this);
+    this.receiveFavouritesInfo = this.receiveFavouritesInfo.bind(this);
   }
 
-  render(){
+  receiveSearchTerm(term) {
+    this.setState({
+      searchTerm: term
+    });
+  }
+
+  fetchMovies() {
+    if (this.state.searchTerm === "") {
+      this.setState({
+        movieArray: []
+      });
+    } else {
+      fetch(
+        `http://www.omdbapi.com/?s=${this.state.searchTerm}&page=${
+          this.state.currentPage
+        }&apikey=${this.state.apiKey}`
+      )
+        .then(response => response.json())
+        .then(body => {
+          this.setState({
+            pagesObject: body,
+            movieArray: body.Search
+          });
+        });
+    }
+  }
+
+  componentDidMount() {
+    this.setState(
+      {
+        favouritesArray:
+          localStorage.getItem("favourites") === null
+            ? []
+            : JSON.parse(localStorage.getItem("favourites"))
+      }
+      // localStorage.clear()
+    );
+  }
+
+  receiveCurrentPage(page) {
+    this.setState(
+      {
+        currentPage: page
+      },
+      () => this.fetchMovies()
+    );
+  }
+
+  receiveFavouritesInfo(id, favourite) {
+    if (
```

Comment:
The logic in this function could be simplified. If you calculate the new favourites array for each condition and store in a variable, you can then set it in state and local storage after. It will save you having to repeat setState and localStorage.setItem